templates/item/preview.macros.j2

```diff
@@ -31,24 +32,35 @@
   Display color palettes based on JSON map output.
 #}
 {% macro color_palette(data) %}
-<div data-sassdoc="color-palette">
-  {% for name, value in data %}
-    <div data-sassdoc-color="{{ name }}">
-      <div data-sassdoc="color-swatch">
-        <div data-sassdoc="transparency-grid"></div>
-        <div data-sassdoc="color-overlay" style="background-color: {{ value.output }};"></div>
-      </div>
-      <div data-sassdoc="color-info">
-        <strong data-sassdoc="color-name">{{ name }}</strong>
-        {% for type, color in value %}
-          <div data-sassdoc="color-value">
-            {{ color }}
+  {% call show.item_section('color-palette', 'Color Preview') %}
+    <div class="color-palette">
+      {% for value, items in data|list|groupby('value') %}
```

Comment:
This is much nicer than my hacky loop - thanks! :)

Reply:
Agreed, and it preserves the original ordering (which is a nice side-effect, I think).